pkg/segment/test/segment_test.go

```diff
@@ -140,6 +258,145 @@ func TestRemoveRandom(t *testing.T) {
 	}
 }
 
+func TestAddRandomRemoveRandomHalfWithMerge(t *testing.T) {
+	var s Set
+	order := randIntervalPermutation(testSize * 2)
+	order = order[:testSize]
+	for i, j := range order {
+		if !s.Add(Range{j, j + 10}, j+valueOffset) {
```

Comment:
The 10 here is meaningful because it's the gap in the randIntervalPermutation. But one would just have to guess that. Maybe make this a const above randIntervalPermutation and use everywhere? E.g. const randIntervalGap = 10